gobblin-service/src/main/java/org/apache/gobblin/service/modules/utils/FlowCompilationValidationHelper.java

```diff
@@ -187,9 +194,59 @@ public Optional<Dag<JobExecutionPlan>> validateAndHandleConcurrentExecution(Conf
    * @param allowConcurrentExecution
    * @return true if the {@link FlowSpec} allows concurrent executions or if no other instance of the flow is currently RUNNING.
    */
-  private boolean isExecutionPermitted(FlowStatusGenerator flowStatusGenerator, String flowGroup, String flowName,
-      boolean allowConcurrentExecution, long flowExecutionId) {
-    return allowConcurrentExecution || !flowStatusGenerator.isFlowRunning(flowName, flowGroup, flowExecutionId);
+  private boolean isExecutionPermitted(String flowGroup, String flowName, boolean allowConcurrentExecution)
+      throws IOException {
+    return allowConcurrentExecution || !isFlowRunning(flowGroup, flowName, dagManagementStateStore);
+  }
+
+  /**
+   * Returns true if any previous execution for the flow determined by the provided flowGroup, flowName is running.
+   * We ignore the execution that has the provided flowExecutionId. We also ignore the flows that are running beyond
```

Comment:
doesn't look like `flowExecutionId` is actually provided

Reply:
oh yea, i missed this explaining about this.
So, I do not understand why the previous code in `FlowStatusGenerator::isFlowRunning` ignores existing flow statuses of the current `flowExecutionId`. Shouldnt we skip the job if the a flow with the current flowExecutionId is already running? @Will-Lo  you may have some idea.

Reply:
We need to ignore the current flowExecutionId because other hosts may have already come to a decision with the current processed flow status. Since in the multileader world, the hosts can become desyncronized it is important that they come to the same decision, otherwise one host can set a dag as RUNNING then another host which is running late will check the current status, and decide that the dag should FAIL due to already being set as RUNNING.

Reply:
Okay, thanks for the explanation. I think with this in mind, it looks ok to consider all execution ids including the current one. This method will be called only by the lease owner and that is supposed to start the flow, if it does, but somehow fails to finish the lease (service restart, or anything), it is expected for the next lease owner to understand that the flow is launched and not re-launch it. cc @phet

Reply:
added a comment with some more explanantion

Reply:
what about re-attemps to process a launch event that was only partially successful?  until processing succeeds fully, a lease is available for the same `DagAction`.

from `LaunchDagProc::initialize`:
```
  protected Optional<Dag<JobExecutionPlan>> initialize(DagManagementStateStore dagManagementStateStore)
      throws IOException {
    try {
      FlowSpec flowSpec = dagManagementStateStore.getFlowSpec(FlowSpec.Utils.createFlowSpecUri(getDagId().getFlowId()));
      flowSpec.addProperty(ConfigurationKeys.FLOW_EXECUTION_ID_KEY, getDagId().getFlowExecutionId());
      Optional<Dag<JobExecutionPlan>> dag = this.flowCompilationValidationHelper.createExecutionPlanIfValid(flowSpec).toJavaUtil();
      if (dag.isPresent()) {
        dagManagementStateStore.addDag(dag.get());
      }
      return dag;
  ...
```
if the prior attempt got to `dagManagementStateStore.addDag` and then failed, wouldn't the subsequent exec never get past `flowCompilationValidationHelper.createExecutionPlanIfValid`, unless this method (called within) agrees to skip past that same (currently validating) flowExecId?

Reply:
in the comment, please elaborate on:
> We ignore the execution that has the provided flowExecutionId.
to talk about the re-attempt of the same `DagAction` as described in the discussion here

Reply:
added